staging/publishing/rules.yaml

```diff
@@ -877,6 +877,22 @@ rules:
         branch: release-1.14
       - repository: component-base
         branch: release-1.14
+- destination: kubectl
```

Comment:
We should not update the publishing rules, yet. There's still some code in our kubectl repo which does not exist in the main k/k code. I want to ensure we bring those back but deprecated and not to break people. Also, the publishing should happen not sooner than when we're close to finishing the move.